depthai_ros_driver/include/depthai_ros_driver/depthai_common.hpp

```diff
@@ -0,0 +1,274 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef DEPTHAI_ROS_DRIVER__DEPTHAI_COMMON_HPP
+#define DEPTHAI_ROS_DRIVER__DEPTHAI_COMMON_HPP
+
+
+// Common ros and ros2 api
+#include <cv_bridge/cv_bridge.h>
+
+// relevant 3rd party includes
+#include <depthai/device.hpp>
+#include <depthai/host_data_packet.hpp>
+#include <depthai/nnet/nnet_packet.hpp>
+#include <depthai/pipeline/cnn_host_pipeline.hpp>
+
+// std includes
+#include <algorithm>
+#include <array>
+#include <memory>
+#include <string>
+#include <variant>
+
+#if defined(USE_ROS2)
+  #include <std_msgs/msg/float32.hpp>
+  #include <sensor_msgs/msg/image.hpp>
+  #include <depthai_ros_msgs/srv/trigger_named.hpp>
+  #include <depthai_ros_msgs/msg/objects.hpp>
+  #include <depthai_ros_msgs/msg/auto_focus_ctrl.hpp>
+  #include <camera_info_manager/camera_info_manager.hpp>
+  #include <rclcpp/node.hpp>
+
+using ImageMsg = sensor_msgs::msg::Image;
+using CompressedImageMsg = sensor_msgs::msg::CompressedImage;
+using ObjectsMsg = depthai_ros_msgs::msg::Objects;
+using ObjectMsg = depthai_ros_msgs::msg::Object;
+using CameraInfoMsg = sensor_msgs::msg::CameraInfo;
+using HeaderMsg = std_msgs::msg::Header;
+using ROSNodeHandle = rclcpp::Node*;
+
+using ObjectPubPtr = rclcpp::Publisher<ObjectMsg>::SharedPtr;
+using ObjectsPubPtr = rclcpp::Publisher<ObjectsMsg>::SharedPtr;
+using ImagePubPtr = rclcpp::Publisher<ImageMsg>::SharedPtr;
+using ComImagePubPtr = rclcpp::Publisher<CompressedImageMsg>::SharedPtr;
+using StreamPub = std::variant<ObjectsPubPtr, ObjectPubPtr,
+    ImagePubPtr, ComImagePubPtr>;
+using CameraInfoPub =
+  rclcpp::Publisher<sensor_msgs::msg::CameraInfo>::SharedPtr;
+
+using RosTime = rclcpp::Time;
+using RosDuration = rclcpp::Duration;
+
+#else
+  #include <std_msgs/Float32.h>
+  #include <sensor_msgs/Image.h>
+  #include <depthai_ros_msgs/AutoFocusCtrl.h>
+  #include <depthai_ros_msgs/Objects.h>
+  #include <depthai_ros_msgs/TriggerNamed.h>
+  #include <camera_info_manager/camera_info_manager.h>
+
+using ImageMsg = sensor_msgs::Image;
+using CompressedImageMsg = sensor_msgs::CompressedImage;
+using ObjectsMsg = depthai_ros_msgs::Objects;
+using ObjectMsg = depthai_ros_msgs::Object;
+using CameraInfoMsg = sensor_msgs::CameraInfo;
+using HeaderMsg = std_msgs::Header;
+
+using ROSNodeHandle = ros::NodeHandle;
+using StreamPub = std::unique_ptr<ros::Publisher>;
+using CameraInfoPub = std::unique_ptr<ros::Publisher>;
```

Comment:
Why are ROS1 types unique_ptr and ROS2 types shared_ptr? That seems quite weird and flaky

Reply:
ros2 `create_publisher()` returns a `shared_ptr`. and the previous impl of ros1 uses `unique_ptr`. for consistency, I can switch all pubs to shared_ptr.